rust/worker/src/system/wrapped_message.rs

```diff
@@ -31,25 +70,17 @@ where
 }
 
 #[async_trait]
-impl<C, M> HandleableMessage<C> for Option<M>
+impl<C, M> HandleableMessage<C> for Option<MessageWithReplyChannel<M, C::Result>>
 where
     C: Component + Handler<M>,
     M: Debug + Send + 'static,
 {
     async fn handle(&mut self, component: &mut C, ctx: &ComponentContext<C>) -> () {
         if let Some(message) = self.take() {
-            component.handle(message, ctx).await;
+            let result = component.handle(message.message, ctx).await;
+            if let Some(reply_channel) = message.reply_channel {
+                reply_channel.send(result).unwrap();
```

Comment:
should we instead ignore errors here?

Reply:
I think that makes sense, as a caller could drop the reply channel receiver and that shouldn't affect the callee

Reply:
lets .expect() for now with an explanation